src/Orchard/Tasks/Locking/Services/Lock.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Orchard.Logging;
+
+namespace Orchard.Tasks.Locking.Services {
+
+    /// <summary>
+    /// Provides a database driven implementation of <see cref="IDistributedLock" />
+    /// </summary>
+    public class Lock : IDistributedLock {
```

Comment:
I agree. `Lock` is too generic and doesn't convey the fact that it is a distributed lock. Will update.